### ISSUE TITLE
`bash.exe` no longer recognizes wsl command surrounded by single quotes.

### ISSUE THREAD

**Issue by gnaggnoyil** [microsoft/WSL#5932]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.18363.1082]
Your Distribution version: Archlinux
Whether the issue is on WSL 2 and/or WSL 1: Consistently occurs on WSL 1, no idea if it affects WSL 2 or not.
```

# Steps to reproduce

* Start `cmd.exe`
* Run command `bash.exe -c 'echo aaa'`

#  Expected behavior

```
C:\>bash.exe -c 'echo aaa'
aaa
```

# Actual behavior

```
C:\>bash.exe -c 'echo aaa'
aaa': -c: line 0: unexpected EOF while looking for matching `''
aaa': -c: line 1: syntax error: unexpected end of file
```

# Additional info

The expected behavior is what would happen when running `bash -c 'echo aaa'` and `bash -c "echo aaa"` under Version 10.0.18363.10016 (aka Win10 1909 before KB4576484) and running `bash -c "echo aaa"` under current version of Win10.

**Comment by WSLUser** [comment]
Deprecated. Use `wsl.exe` instead.

**Comment by benhillis** [comment]
Looks like this is fixed in 19042.

**Comment by gnaggnoyil** [comment]
Nvm, confirmed this issue is fixed in Win10 21H1 19043.1415 already.
